### DISCUSSION TITLE
Query Key containing BigInt

### DISCUSSION THREAD

**Discussion by JesusTheHun** [TanStack/query#2450] [General] [answered]
My app uses native `BigInt`. Some of the object that carry them are passed as elements of the Query Key.

It looks like internally `react-query` uses `JSON.stringify` on the key, which leads to an error since `JSON` doesn't know how to deal with `BigInt`.

Shall I open a bug report for that, is it considered a feature request ?

**Comment by TkDodo** [comment, marked as answer]
Elements in the queryKey must be json serializable. You van implement your own queryHashFn for a custom implementation if you need more.

**Reply by JesusTheHun** [reply]
Oh I didn't see I could write my own implementation. Thx !
